### PR TITLE
RavenDB-14781 Ensure certificate serial number is exactly 20 bytes in…

### PR DESCRIPTION
**Pull request by iftahbe**
… length

### CODE REVIEW

File: src/Raven.Server/Utils/CertificateUtils.cs [hunk 1]
```diff
@@ -186,8 +186,21 @@ public static X509Certificate2 CreateSelfSignedExpiredClientCertificate(string c
...
+            // Serial Number - we must generate a positive number, so retrying if it's not
+            BigInteger serialNumber;
+            var retries = 10;
+
+            while (true)
+            {
+                var serialNumberBytes = new byte[20];
+                random.NextBytes(serialNumberBytes);
+                serialNumber = new BigInteger(serialNumberBytes);
+                if (serialNumber.SignValue > 0)
```
Comment (ayende) [inline src/Raven.Server/Utils/CertificateUtils.cs]: Huh? Why do we need this like that?

Comment (iftahbe) [reply src/Raven.Server/Utils/CertificateUtils.cs]: If the generated bytes form a negative number, certificateGenerator.SetSerialNumber() will throw, so I'm making sure random gives us a positive number.

Comment (ayende) [reply src/Raven.Server/Utils/CertificateUtils.cs]: Then why regenerate the number? flip the sign

https://github.com/bcgit/bc-csharp/blob/master/crypto/src/math/BigInteger.cs#L721

Comment (iftahbe) [reply src/Raven.Server/Utils/CertificateUtils.cs]: Ooh that's nice, I missed this method.

**Review by ayende — APPROVED** [review]

**Pull request merged**